src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -221,13 +254,37 @@ protected override void SetImplementationFlagsCore(MethodImplAttributes attribut
 
         public override MethodInfo GetGenericMethodDefinition() => !IsGenericMethod ? throw new InvalidOperationException() : this;
 
-        public override int GetHashCode()
-            => throw new NotImplementedException();
-
         public override MethodImplAttributes GetMethodImplementationFlags()
             => _methodImplFlags;
 
-        public override ParameterInfo[] GetParameters() => throw new NotImplementedException();
+        public override ParameterInfo[] GetParameters()
+        {
+            // This is called from ILGenerator when Emit(OpCode, ConstructorInfo) when the ctor is
+            // instance of 'ConstructorOnTypeBuilderInstantiation', so we could not throw here even
+            // the type was not baked. Runtime implementation throws when the type is not baked.
+
+            if (_parameterTypes == null)
+            {
+                return Array.Empty<ParameterInfo>();
+            }
+
+            _parameterBuilders ??= new ParameterBuilderImpl[_parameterTypes.Length + 1]; // parameter 0 reserved for return type
+            ParameterInfo[] parameters = new ParameterInfo[_parameterTypes.Length];
+
+            for (int i = 0; i < _parameterTypes.Length; i++)
+            {
+                if (_parameterBuilders[i + 1] == null)
```

Comment:
When is the element `null` in this case?

Reply:
When the parameters created with `parameterTypes` array: https://github.com/dotnet/runtime/blob/ffd43b6d9258e03543dac7f7e5b9ff54e1e7a3b6/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs#L53-L60
but not explicitly defined with `DefineParameterCore(...)`